pkg/oauth/interactive.go

```diff
@@ -15,9 +15,121 @@
+					</svg>
+				</a>
+			</div>
+			<div>
+				<div class="title">
+					<span class="sigstore">sigstore </span>
+					<span>authentication successful!</span>
+				</div>
+				{{ if .Autoclose -}}
+				<small name="autoclose"></small>
+				<noscript>
+					<div class="content">
+						<span>You may now close this page.</span>
+					</div>
+				</noscript>
+				{{- else -}}
+				<div class="content">
+					<span>You may now close this page.</span>
+				</div>
+				{{- end }}
+			</div>
+			<div class="anchor">
+				<div class="links">
+					<a href="https://sigstore.dev/" class="link login"><span class="sigstore">sigstore</span> home <span class="arrow">→</span></a>
+					<a href="https://docs.sigstore.dev/" class="link login"><span class="sigstore">sigstore</span> documentation <span class="arrow">→</span></a>
+					<a href="https://blog.sigstore.dev/" class="link"><span class="sigstore">sigstore</span> blog <span class="arrow">→</span></a>
+				</div>
+			</div>
+		</div>
+		<script>
+			document.getElementById("favicon").setAttribute("href", "data:image/svg+xml," + encodeURIComponent(document.getElementById("logo").outerHTML));
+		</script>
+
+		{{ if .Autoclose -}}
+		<script>
+			var timeout = {{ .Timeout }};
+			setTimeout(function() { this.close(); }, timeout*1000);
+			setInterval(function() {
+				timeout--;
+				document.getElementsByName("autoclose")[0].innerHTML = "This page will close automatically in " + timeout + " seconds...";
+				if (timeout <= 0) {
+					document.getElementsByName("autoclose")[0].innerHTML = "This page will close now, thank you!";
+				}
+			}, 1000);
+		</script>
+		{{- end }}
+	</body>
+</html>
+`
+	// Parse the template
+	tmpl, err := template.New("success").Parse(successTemplate)
+	if err != nil {
+		return "", fmt.Errorf("error parsing success template: %w", err)
+	}
+	// Pass autoclose and timeout to the template
+	data := struct {
+		Autoclose bool
+		Timeout   int
+	}{
+		autoclose,
+		timeout,
+	}
+	var htmlPage bytes.Buffer
+	if err := tmpl.Execute(&htmlPage, data); err != nil {
+		return "", fmt.Errorf("error executing template: %w", err)
+	}
+	return htmlPage.String(), nil
+}
+
 const (
-	// InteractiveSuccessHTML is the page displayed upon success when using a web browser during an interactive Oauth token flow.
-	InteractiveSuccessHTML = `<!DOCTYPE html>
+	// DefaultInteractiveSuccessHTML is the page displayed upon success when using a web browser during an interactive Oauth token flow.
+	DefaultInteractiveSuccessHTML = `<!DOCTYPE html>
```

Comment:
This may break downstream users that are using on the old var. If you want to rename this, perhaps we should deprecate + alias this instead?

e.g.

```go
const (
  // Deprecated: ...
  InteractiveSuccessHTML = ...
)

var (
  DefaultInteractiveSuccessHTML = InteractiveSuccessHTML
)
```

Alternatively we could only deprecate and point them to GetInteractiveSuccessHTML.

Reply:
Sure, I made that change [based on the last review](https://github.com/sigstore/sigstore/pull/1618#discussion_r1485233325) but wasn't super happy with how I ended up refactoring it. Would you like me to implement your suggestion as well?

Reply:
I agree on avoiding breaking changes. I am happy with the change to switch from panic to error too.